haskell-src/Concordium/Genesis/Data/P2.hs

```diff
@@ -0,0 +1,67 @@
+-- |This module defines the genesis data fromat for the 'P1' protocol version.
+module Concordium.Genesis.Data.P2 where
+
+import Data.Serialize
+
+import Concordium.Common.Version
+import qualified Concordium.Crypto.SHA256 as Hash
+import Concordium.Genesis.Data.Base
+import Concordium.Types
+
+-- |Genesis data for the P2 protocol version.
+newtype GenesisDataP2
+    = GenesisDataP2 { unGenesisDataP2 :: RegenesisData }
```

Comment:
I have added this now by essentially copying the P1 one.

I thought that we should instead have just serialized state instead of the genesis parameters, but that causes more problems since the rest of the types/infrastructure is not ready, and I do not have time to spend refactoring things for no clear gain at present.

If this ever becomes an issue it can be revisited in the future.

Reply:
I'll push the update to the genesis tool later tonight.

Reply:
Done.